### PR TITLE
Cached servers using json file

### PR DESCRIPTION
**Pull request by AmyrAhmady**
This PR adds ability to retrieve server data through API from a JSON file, instead of database; This helps a with ton of DB processing and going out of limit every month, meaning you ONLY need database data (to add or update) on each scraper worker run (every 30 minutes now)

I'm doing this because lately due to samp lists going down, we are getting a lot of traffics by users using our list seriously now

### CODE REVIEW

**Review by karimcambridge — APPROVED** [review]

File: app/transports/api/servers/api.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ func Build() fx.Option {
 
 			rtr.Get("/{address}", s.get)
 			rtr.Get("/", s.list)
+			rtr.Get("/full", s.fulllist)
```
Comment (Southclaws) [inline app/transports/api/servers/api.go:63]: don't we want to apply the cache to the main /servers endpoint too? that's the one I think most people use

**Review by Southclaws — APPROVED** [review]
looks good, my comments aren't blocking, let's get this released ASAP and maybe revisit with a longer term solution

File: app/transports/api/servers/api.go (hunk 1 above)
Comment (AmyrAhmady) [reply app/transports/api/servers/api.go:63]: main/old ones use cached files as well

File: app/services/serverworker/serverworker.go [hunk 2]
```diff
@@ -2,8 +2,10 @@ package serverworker
 
 import (
 	"context"
+	"os"
 	"time"
 
+	"github.com/goccy/go-json"
```
[the remark this answers is not in the document]
Comment (AmyrAhmady) [reply app/services/serverworker/serverworker.go:8]: huge json file processing due to big server list with full data (including rules)

**Pull request merged**